sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/DriverRegistry.scala

```diff
@@ -32,6 +32,9 @@ import org.apache.spark.util.Utils
  */
 object DriverRegistry extends Logging {
 
+  // Initialize DriverManager first to prevent potential deadlocks between DriverManager and Driver
```

Comment:
We need to say more about why this can avoid deadlock.

Reply:
we can copy something from the storm PR: https://github.com/apache/storm/pull/2134/files